storage/watchable_store.go

```diff
@@ -287,6 +291,15 @@ func (s *watchableStore) syncWatchers() {
 		s.unsynced.delete(w)
 	}
 
+	// bring all un-notified watchers to synced.
```

Comment:
So does this kind of resolve `// TODO: handle the full unsynced watchers...` lines above?